pallets/gear/src/lib.rs

```diff
@@ -1118,21 +1121,10 @@ pub mod pallet {
         /// test (`schedule::tests::instructions_backward_compatibility`)
         pub(crate) fn reinstrument_code(
             code_id: CodeId,
+            original_code: Vec<u8>,
             schedule: &Schedule<T>,
         ) -> Result<InstrumentedCode, CodeError> {
-            debug_assert!(T::CodeStorage::get_code(code_id).is_some());
-
-            // By the invariant set in CodeStorage trait, original code can't exist in storage
-            // without the instrumented code
-            let original_code = T::CodeStorage::get_original_code(code_id).unwrap_or_else(|| {
-                let err_msg = format!(
-                    "reinstrument_code: failed to get original code, while instrumented exists. \
-                    Code id - {code_id}"
-                );
-
-                log::error!("{err_msg}");
-                unreachable!("{err_msg}")
-            });
```

Comment:
Could you clarify the purpose of calling this code before each `reinstrument_code` call?

Reply:
We need bytes of the original code to call instrumented code generation, so this function reads the original code from the repository itself. 
I thought this was an unnecessary feature, so I did the reading of the original code before calling this function, and now I pass the bytes of original code to it from outside. 
I did the same with writing the instrumented code and metadata: I moved it outside of this function.

Reply:
actually, why? at the moment of reading meta, you hasn't read the original code yet, so it looks ok for me to hide it here